crates/contract/src/lib.rs

```diff
@@ -3071,4 +3112,46 @@ mod tests {
             .build());
         contract.remove_update_vote();
     }
+
+    #[test]
+    pub fn test_update_votes_cleared_after_resharing() {
```

Comment:
The name of this function is mis-leading: it does not test whether votes are cleared after a resharing, but if `PrposodeUpdates::remove_non_participant_votes()` works.

This test doesn't need to be in this file, as it does not test a contract method. It can be massively simplified and moved to `votes.rs`